BusinessLogic/Entities/EventDispatcher.cs

```diff
@@ -41,6 +42,29 @@ private EventDispatcher()
         public void Register(Subscription subscription)
         {
             Log.Debug("EventDispatcher.Register");
+
+
```

Comment:
Let's add some documentation here to ensure that it is clear what is happening

```suggestion
            // If there is already a synchronous subscription then raise an exception
```

Reply:
fixed too